src/Microsoft.Data.Sqlite/Interop/NativeMethods.cs

```diff
@@ -2,12 +2,34 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Reflection;
 using System.Runtime.InteropServices;
+using Microsoft.Data.Sqlite.Utilities;
 
 namespace Microsoft.Data.Sqlite.Interop
 {
     internal static partial class NativeMethods
     {
+        static NativeMethods()
```

Comment:
:smirk: We could call the native APIs directly...
Or just use a flag to detect re-entrance and no-op.